### PR TITLE
Add a timeout for GCE cluster initialization

### PR DESCRIPTION
**Pull request by ixdy**
It appears that successful GCE-based clusters start up within a few seconds. Rather than waiting for the long Jenkins timeout for clusters which will never start, fail fast with a conservative default timeout of 5 minutes.

This should help with the pull request Jenkins, where we often wait 45m on a cluster which will never work.

@kubernetes/goog-testing

### CODE REVIEW

File: cluster/gce/util.sh [hunk 1]
```diff
@@ -736,12 +737,18 @@ function kube-up {
...
-
+  local start_time=$(date +%s)
   until curl --cacert "${CERT_DIR}/pki/ca.crt" \
           -H "Authorization: Bearer ${KUBE_BEARER_TOKEN}" \
           ${secure} \
           --max-time 5 --fail --output /dev/null --silent \
           "https://${KUBE_MASTER_IP}/api/v1/pods"; do
+      local elapsed=$(($(date +%s) - ${start_time}))
+      if [[ ${elapsed} -gt ${KUBE_CLUSTER_INITIALIZATION_TIMEOUT} ]]; then
+          echo -e "${color_red}Cluster failed to initialize within" \
```
Comment (quinton-hoole) [inline cluster/gce/util.sh]: nit: This should ideally go to stderr.

**Comment by quinton-hoole** [comment]
LGTM.

**Comment by quinton-hoole** [comment]
Jenkins failure looks like a flake.

File: cluster/gce/util.sh (hunk 1 above)
Comment (ixdy) [inline cluster/gce/util.sh]: this was mostly due to cargo-culting other code in this file. I've now updated the other errors which should've gone to stderr, too.

**Comment by quinton-hoole** [comment]
Looks even more Good To Me :-)

**Comment by ixdy** [comment]
@quinton-hoole can I get a re-lgtm?

**Comment by ixdy** [comment]
actually, I'm just going to self-label it

**Comment by quinton-hoole** [comment]
Yup, still looks good to me :-)

**Pull request merged**